src/layout/rect.rs

```diff
@@ -462,6 +499,84 @@ mod tests {
         );
     }
 
+    #[test]
```

Comment:
Can you please add a test for i32::MIN and i32::MAX? There's likely no bug, but this prevents any math errors in any future implementation changes.

Reply:
Added two test, that should cover the `ResizeOffset::new` function as well